doc/man3/OSSL_PARAM_int.pod

```diff
@@ -241,17 +241,18 @@ will be assigned the size the parameter's I<data> buffer should have.
 OSSL_PARAM_get_utf8_string() retrieves a UTF8 string from the parameter
 pointed to by I<p>.
 The string is stored into I<*val> with a size limit of I<max_len>,
-which must be large enough to accomodate a terminating NUL byte,
-otherwise this function will fail.
-If I<*val> is NULL, memory is allocated for the string and I<max_len>
-is ignored.
+which must be large enough to accomodate the string. A terminating NUL byte
+is added only if the buffer is longer than the string length otherwise the
+string will not be NUL terminated.
```

Comment:
This change doesn't reflect that the function _does_ fail if there isn't space for a NUL byte:

https://github.com/openssl/openssl/blob/4ccad35756dfa9df657f3853810101fa9d6ca525/crypto/params.c#L1135-L1136

Reply:
Sadly, `OSSL_PARAM_get_utf8_string()` is not the function called by
* `EVP_PKEY_get_group_name(3)`,
* which calls `EVP_PKEY_get_utf8_string_param()`,
* which in turn calls `EVP_PKEY_get_params()`,
* which calls `evp_keymgmt_get_params()`, ...
* 
where ultimately there does not appear to be a similar check.

Reply:
Perhaps the behaviour of EVP_PKEY_get_group_name is actually incorrect, not just (previously) under-documented.

Reply:
All of that tells me that this documentation change should have been made for `OSSL_PARAM_set_utf8_string()`.  Regardless of your comments, the documentation change doesn't reflect what `OSSL_PARAM_get_utf8_string()` does.

I'll submit a fixup PR promptly

Reply:
See #16318

Reply:
I would say that `EVP_PKEY_get_group_name()` is underdocumented, then.  The documentation does say that `*gname_len` will be assigned the actual length.

Reply:
> I would say that `EVP_PKEY_get_group_name()` is underdocumented, then. The documentation does say that `*gname_len` will be assigned the actual length.

The original words said `size` and did not specify whether it was with or without the NUL.  GIven that the input size was with the NUL, I'd expected to output size to likewise include the NUL but it does not.  Hence the open issue.

And the other surprising part was that the function succeeds and returns a a value that is not NUL-terminated.  Perhaps this is a bug?

Reply:
Does #16334 alleviate your concern, @vdukhovni ?